### ISSUE TITLE
IAP (item marketplace)

### ISSUE THREAD

**Issue by dev816** [YoYoGames/GMEXT-Steamworks#22]
How to use IAP (or item marketplace) for free game in this extansion? Thanks

**Comment by DiasFranciscoA** [comment]
This is not possible because there is no implementation for the **ISteamMicroTxn** interface on our extension.

I've turned this into a **Feature Request** that can be voted and we will consider that at a later date.

**Comment by dev816** [comment]
Personally, I and at least 50 other people from the community still need it, + I saw on Reddit that they also wrote about it. Please, I think it's very useful, or tell me where to start to do it yourself

**Comment by DiasFranciscoA** [comment]
Upon further investigation, it appears that the Microtransactions feature utilizes a WebAPI, which means it primarily relies on http_requests and listens for http async responses.

While awaiting further assistance, you may find it helpful to review the documentation to familiarize yourself with the usage of the available endpoints. You can find the documentation for this feature at the following link: https://partner.steamgames.com/doc/webapi/ISteamMicroTxn.

**Comment by SanyaShanin** [comment]
> Upon further investigation, it appears that the Microtransactions feature utilizes a WebAPI, which means it primarily relies on **http_requests** and listens for **http async responses**.
> 
> While awaiting further assistance, you may find it helpful to review the documentation to familiarize yourself with the usage of the available endpoints. You can find the documentation for this feature at the following link: https://partner.steamgames.com/doc/webapi/ISteamMicroTxn.

But everyone needs to remember that this call cannot be used in the game, because it contains a secret api key. Therefore, those who want to do this need to make a dedicated server that would store this key and access the steam api

**Comment by msviator** [comment]
Is there an update on this feature? Currently, there is no IAP function, so I am replacing it with the DLC function. However, DLC can only purchase it once, so we need to keep increasing the same DLC for users who want to buy it repeatedly.

**Comment by kingsushi001** [comment]
I too would like this 🥲

**Comment by dev816** [comment]
will they do it in 2025?

**Comment by Matbart1** [comment]
This issue should be archived. The answer has already been stated. No, there will not be 'native' functionality for this, because it's available through HTTPS requests. These features require knowledge of servers, and endpoint accessing.

That is to say, this is already accessible, and no development needs to be done on GMEXT's side of things.

**Comment by dev816** [comment]
[Matbart1](https://github.com/Matbart1) let them make an example of how to do this?

**Comment by Foxtrio** [comment]
@dev816 Steamwork's client SDK does not have this ability. Steam expects you to handle the middle-men role for the communication of client <> Steam. Therefore, you must create a webserver of your own and handle it there.

That being said, you might want to check the following for implementation.
https://github.com/jasielmacedo/steam-microtransaction-api

**Comment by dev816** [comment]
@Foxtrio Exactly, I don't want to reinvent the wheel — I want an official template from the creators that I can adapt for my project.

**Comment by dev816** [comment]
Gm is rip? Why are we still using 2024.13 in 2025? What's going on with you? Is it that difficult to add documentation? Why do I have to wait years for an answer to my question?
